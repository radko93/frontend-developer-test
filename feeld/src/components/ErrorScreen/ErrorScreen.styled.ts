import styled from 'styled-components/native'
import { Ionicons } from '@expo/vector-icons'
import colors from '../../constants/colors'
import { responsiveDPSize } from '../../utils/responsive'

export const RetryIcon = styled(Ionicons)`
  color: ${colors.grey};
  margin-bottom: ${responsiveDPSize(30)};
`

export const Label = styled.Text`
  text-align: center;
  color: ${colors.grey};
`

export const Container = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
`
