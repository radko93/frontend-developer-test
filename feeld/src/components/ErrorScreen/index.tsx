import React from 'react'
import { TouchableOpacity } from 'react-native'
import i18n from '../../translations'
import { Container, Label, RetryIcon } from './ErrorScreen.styled'
import { responsiveDPSize } from '../../utils/responsive'

export type Props = {
  onRetry: () => void
}

const OfflineRetry = ({ onRetry }: Props) => (
  <Container>
    <TouchableOpacity testID="retry-button" onPress={onRetry}>
      <RetryIcon name="md-refresh" size={responsiveDPSize(50)} />
    </TouchableOpacity>
    <Label>{i18n.t('offlineScreen:errorMessage')}</Label>
  </Container>
)

export default OfflineRetry
