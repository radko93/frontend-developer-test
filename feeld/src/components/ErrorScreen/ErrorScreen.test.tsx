import React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import ErrorScreen from '.'

const props = {
  onRetry: jest.fn(),
}

describe('ErrorScreen Component', (): void => {
  it('Renders ErrorScreen with default props', (): void => {
    const wrapper = render(<ErrorScreen {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Should fire onPress event', (): void => {
    const wrapper = render(<ErrorScreen {...props} />)
    const { getByTestId } = wrapper
    fireEvent.press(getByTestId('retry-button'))
    expect(props.onRetry).toBeCalled()
  })
})
