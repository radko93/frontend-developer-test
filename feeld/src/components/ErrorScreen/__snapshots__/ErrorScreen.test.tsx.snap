// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorScreen Component Renders ErrorScreen with default props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="retry-button"
  >
    <Text />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#989898",
          "textAlign": "center",
        },
      ]
    }
  >
    An error occured, please try again.
  </Text>
</View>
`;
